staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -762,7 +763,8 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 
 			data, _, err := s.transformer.TransformFromStorage(ctx, kv.Value, authenticatedDataString(kv.Key))
```

Comment:
Should we consider wrapping TransformFromStorage errors?

I think that the current error that we expose in the status makes it hard to understand what's happening:
```
Type:    "UnexpectedServerResponse",
Message: "cipher: message authentication failed",
```

Reply:
This is already part of the `StatusDetails.Causes`, right?

Reply:
Yes but I am wondering if from a user perspective it would be easy to understand that `cipher: message authentication failed` means that the server couldn't decrypt the secret because the key couldn't be retrieved. But maybe having it said explicitly in the Status.Message is clear enough?

Reply:
Nevermind, thinking about it again, I've been too focused on the reporting aspect, which should've already been covered by the KMS v2 effort. As you mentioned in another comment, this kep is mainly about unblocking scenario where the data can't be deleted and recreated. So reporting the message as is is fine.